### PR TITLE
Allow for default locale to be configured in constructor

### PR DESCRIPTION
**Pull request by jbevis**

### CODE REVIEW

**Comment by chrisdmacrae** [comment]
@jbevis let's update this api to have this interface/type, and also create an `const` alias called `i18nClient` and export it as well:

```
export interface LocalizationApiOptions {
  locales: LocaleList;
  defaultLocale?: Locale;
  imgMap?: ImgMap
}

export class LocalizationApi {
  constructor(private options: LocalizationApiOptions) {}

  ...
}
```

File: src/localizationApi/LocalizationAPI.ts [hunk 1]
```diff
@@ -33,6 +41,7 @@ export interface Locale {
 export interface LocalizationApiOptions {
```
Comment (chrisdmacrae) [inline src/localizationApi/LocalizationAPI.ts:41]: @jbevis let's update this api to have this interface/type, and also create an `const` alias called `i18nClient` and export it as well:

```
export interface LocalizationApiOptions {
  locales: LocaleList;
  defaultLocale?: Locale;
  imgMap?: ImgMap
}

export class LocalizationApi {
  constructor(private options: LocalizationApiOptions) {}

  ...
}
```

**Review by chrisdmacrae — CHANGES_REQUESTED** [review]

File: src/localizationApi/LocalizationAPI.ts (hunk 1 above)
Comment (chrisdmacrae) [reply src/localizationApi/LocalizationAPI.ts:41]: @jbevis @logan-anderson we also probably want to "soft deprecate" `LocalizationApi` as the nom de jour for API classes are `{{ THING }}Client`, e.g:

```
apis: {
  github: new GithubClient(...),
  contentful: new ContentfulClient(...),
  i18n: new i18nClient(...)
}
```

Comment (jbevis) [reply src/localizationApi/LocalizationAPI.ts:41]: @chrisdmacrae @logan-anderson 

Whomever has some time, I can't seem to get the requested changes to work. Errors out on a number of rather obscure (to me) typescript errors. Would appreciate a second set of eyes.

```Type 'Locale | undefined' is not assignable to type 'Locale'.
  Type 'undefined' is not assignable to type 'Locale'```

Comment (jbevis) [reply src/localizationApi/LocalizationAPI.ts:47]: Updated the API with the requested changes.

**Comment by jbevis** [comment]
@chrisdmacrae ready for next round of review.

**Comment by jbevis** [comment]
closes #25

File: src/localizationApi/LocalizationAPI.ts [hunk 2]
```diff
@@ -23,17 +36,23 @@ export interface CodeAndLabel {
...
   language?: Language;
   region?: Region;
   encoding?: string;
   modifiers?: string[];
 }
 export interface LocalizationApiOptions {
-  localeList?: Locale[];
-  imgMap?: Record<string, any>;
+  localeList: LocaleList;
+  imgMap: ImgMap;
```
Comment (logan-anderson) [inline src/localizationApi/LocalizationAPI.ts:49]: ImgMap should be optional

File: src/localizationApi/LocalizationAPI.ts [hunk 3]
```diff
@@ -23,17 +36,23 @@ export interface CodeAndLabel {
...
+  locale: Locale;
```
Comment (logan-anderson) [inline src/localizationApi/LocalizationAPI.ts:50]: This should probably be optional as well.

**Pull request merged**